File: Dvizh/src/Dvizh.Application/Models/Invite.cs
using Dvizh.Application.Enums;
using Nexus.Application.Core.Interfaces;

namespace Dvizh.Application.Models;

public class Invite : IHasCreatedAt, IHasUpdatedAt, IHasPrimaryIdentifier, IHasExpiresAt
{
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
    public int Id { get; set; }
    public string Code { get; set; } = null!;
    public string Message { get; set; } = null!;
    public string? Description { get; set; }
    public DateTime? ExpiresAt { get; set; }
    public InviteAnswer Answer { get; set; }
    public InviteLanguage Language { get; set; }
    public InviteMascot Mascot { get; set; }
}

File: Dvizh/src/Dvizh.UI/src/types/invite.ts
export enum InviteAnswer {
  Pending = 0,
  Yes = 1,
  No = 2,
}

export enum InviteLanguage {
  Russian = 0,
  Ukrainian = 1,
  English = 2,
}

export enum InviteMascot {
  MochiPeachCat = 0,
  UtyaDuck = 1,
}

export interface Invite {
  id: number
  code: string
  message: string
  description?: string
  expiresAt?: string
  answer: InviteAnswer
  language: InviteLanguage
  mascot: InviteMascot
  createdAt: string
  updatedAt: string
}

export interface PagedResult<T> {
  items: T[]
  totalCount: number
  page: number
  pageSize: number
  totalPages: number
}

export interface DomainError {
  errorCode: string
  errorMessage?: string
  canRetry: boolean
}
